lib/topological_inventory/openshift/collectors_pool.rb

```diff
@@ -0,0 +1,85 @@
+require "config"
+require "topological_inventory/openshift/collector"
+require "topological_inventory/openshift/logging"
+
+module TopologicalInventory::Openshift
+  class CollectorsPool
+    include Logging
+
+    def initialize(config_name, metrics, poll_time: 10)
+      self.collectors  = {}
+      self.collector_threads = {}
+      self.config_name = config_name
+      self.metrics     = metrics
+      self.poll_time   = poll_time
+    end
+
+    def run!
+      loop do
+        reload_config
+
+        remove_old_collectors
+        add_new_collectors
+
+        sleep(poll_time)
+      end
+    end
+
+    def stop!
+      collectors.each_value { |collector| collector.stop }
+
+      # Wait for end of collectors to ensure metrics are stopped after them
+      collector_threads.each { |thread| thread.join }
```

Comment:
Probably want to add some sort of timeout to the join and hard kill the thread if it is exceeded otherwise shutdown can hang.

E.g.: `thread.kill unless thread.join(30)`